### ISSUE TITLE
IndexError on batch translate with auto-detected source language

### ISSUE THREAD

**Issue by std0** [LibreTranslate/LibreTranslate#532]
Hi!
I've got the following exception while trying to send batch translate request with auto-detected source language:

```
2023-11-08 12:54:31 ERROR:libretranslate.app:Exception on /translate [POST]
2023-11-08 12:54:31 Traceback (most recent call last):
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/flask/app.py", line 2525, in wsgi_app
2023-11-08 12:54:31     response = self.full_dispatch_request()
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/flask/app.py", line 1822, in full_dispatch_request
2023-11-08 12:54:31     rv = self.handle_user_exception(e)
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/flask/app.py", line 1820, in full_dispatch_request
2023-11-08 12:54:31     rv = self.dispatch_request()
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/flask/app.py", line 1796, in dispatch_request
2023-11-08 12:54:31     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/libretranslate/app.py", line 283, in func
2023-11-08 12:54:31     return f(*a, **kw)
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/libretranslate/app.py", line 633, in translate
2023-11-08 12:54:31     raise e
2023-11-08 12:54:31   File "/app/venv/lib/python3.10/site-packages/libretranslate/app.py", line 586, in translate
2023-11-08 12:54:31     translator = src_langs[idx].get_translation(tgt_lang)
2023-11-08 12:54:31 IndexError: list index out of range
```

The request that produced the exception:
```
POST /translate HTTP/1.1
Host: localhost:59357
Content-Type: application/json
Content-Length: 117

{
    "q": [
        "hello",
        "world"
    ],
    "source": "auto",
    "target": "es",
    "format": "text"
}
```

The problem appeared after updating to 1.5.0. Therefore, I am guessing that this may be due to the recent [detector replacement](https://github.com/LibreTranslate/LibreTranslate/pull/528).

**Comment by pierotofy** [comment]
Can confirm, this is a bug. `curl -X POST http://localhost:4000/translate -H 'Content-Type: application/json' -d '{"q": ["hello","world"],"source": "auto","target": "es","format": "text"}'`

**Comment by pierotofy** [comment]
Should be fixed with #533, thanks for the report @std0 ! 🙏
